webpack/JobInvocationDetail/JobInvocationActions.js

```diff
@@ -20,3 +30,124 @@ export const getData = url => dispatch => {
 
   dispatch(fetchData);
 };
+
+export const updateJob = jobId => dispatch => {
+  const url = foremanUrl(`/api/job_invocations/${jobId}`);
+  dispatch(
+    APIActions.get({
+      url,
+      key: UPDATE_JOB,
+    })
+  );
+};
+
+export const cancelJob = (jobId, force) => dispatch => {
+  const infoToast = () =>
+    force
+      ? sprintf(__('Trying to abort the job %s.'), jobId)
+      : sprintf(__('Trying to cancel the job %s.'), jobId);
+  const errorToast = response =>
+    force
+      ? sprintf(__(`Could not abort the job %s: ${response}`), jobId)
+      : sprintf(__(`Could not cancel the job %s: ${response}`), jobId);
+  const url = force
+    ? `/job_invocations/${jobId}/cancel?force=true`
+    : `/job_invocations/${jobId}/cancel`;
+
+  dispatch(
+    APIActions.post({
+      url,
+      key: CANCEL_JOB,
+      errorToast: ({ response }) =>
+        errorToast(
+          // eslint-disable-next-line camelcase
+          response?.data?.error?.full_messages ||
+            response?.data?.error?.message ||
+            'Unknown error.'
+        ),
+      handleSuccess: () => {
+        dispatch(
+          addToast({
+            key: `cancel-job-error`,
+            type: 'info',
+            message: infoToast(),
+          })
+        );
+        dispatch(updateJob(jobId));
+      },
+    })
+  );
+};
+
+export const getTask = taskId => dispatch => {
+  dispatch(
+    get({
+      key: GET_TASK,
+      url: `/foreman_tasks/api/tasks/${taskId}`,
+    })
+  );
+};
+
+export const enableRecurringLogic = (
+  recurrenceId,
+  enabled,
+  jobId
+) => dispatch => {
+  const successToast = () =>
+    enabled
+      ? sprintf(__('Recurring logic %s disabled successfully.'), recurrenceId)
+      : sprintf(__('Recurring logic %s enabled successfully.'), recurrenceId);
+  const errorToast = response =>
+    enabled
+      ? sprintf(
+          __(`Could not disable recurring logic %s: ${response}`),
+          recurrenceId
+        )
+      : sprintf(
+          __(`Could not enable recurring logic %s: ${response}`),
+          recurrenceId
+        );
+  const url = `/foreman_tasks/api/recurring_logics/${recurrenceId}`;
+  dispatch(
+    APIActions.put({
+      url,
+      key: CHANGE_ENABLED_RECURRING_LOGIC,
+      params: { enabled: !enabled },
+      successToast,
+      errorToast: ({ response }) =>
+        errorToast(
+          // eslint-disable-next-line camelcase
+          response?.data?.error?.full_messages ||
+            response?.data?.error?.message ||
+            'Unknown error.'
+        ),
+      handleSuccess: () => dispatch(updateJob(jobId)),
+    })
+  );
```

Comment:
@adamruzicka this action doesn't work, even though I think it should, and I'm stuck on it now. I need the `state` of the recurring job to change (`active` <-> `disabled`). Could you please take a look at it if you have time? This is the API (foreman_tasks/api/recurring_logics_controller.rb) I'm trying to use:

```
api :PUT, '/recurring_logics/:id', N_('Update recurring logic')
param :id, :identifier, desc: 'ID of the recurring logic', required: true
param :enabled, :boolean, desc: 'Whether the recurring logic is enabled or disabled.', required: false
def update
     process_response @recurring_logic.update(recurring_logic_params)
end
```

Reply:
Try setting `params` to `{ recurring_logic: { enabled: !enabled } }`, that should do the trick

Reply:
Thank you!